crates/storage/store_db/libmdbx.rs

```diff
@@ -70,6 +70,49 @@ impl Store {
         txn.commit().map_err(StoreError::LibmdbxError)
     }
 
+    fn write_with_txn<T: Table>(
+        txn: &mut libmdbx::orm::Transaction<'_, libmdbx::RW>,
+        key: T::Key,
+        value: T::Value,
+    ) -> Result<(), StoreError> {
+        txn.upsert::<T>(key, value)
+            .map_err(StoreError::LibmdbxError)
+    }
+
+    fn write_batch_with_txn<T: Table>(
+        txn: &mut libmdbx::orm::Transaction<'_, libmdbx::RW>,
+        key_values: impl Iterator<Item = (T::Key, T::Value)>,
+    ) -> Result<(), StoreError> {
+        for (key, value) in key_values {
+            txn.upsert::<T>(key, value)
+                .map_err(StoreError::LibmdbxError)?;
+        }
+
+        Ok(())
+    }
+
+    fn begin(&self) -> Result<libmdbx::orm::Transaction<'_, libmdbx::RW>, StoreError> {
```

Comment:
I believe you'll need to abstract the `Transaction` type to be able to call this while persisting the block. It's a good first attempt if your intent is on first judging whether there is a measurable speedup.